### DISCUSSION TITLE
The API Reference TextEdit needs improvement

### DISCUSSION THREAD

**Discussion by kedicesur** [microsoft/vscode-discussions#1044] [Extension Development QnA] [answered]
I have just noticed that at the [TextEdit](https://code.visualstudio.com/api/references/vscode-api#TextEdit) part of the VS Code API
delete(range: [Range](https://code.visualstudio.com/api/references/vscode-api#Range)): [TextEdit](https://code.visualstudio.com/api/references/vscode-api#TextEdit)
replace(range: [Range](https://code.visualstudio.com/api/references/vscode-api#Range), newText: string): [TextEdit](https://code.visualstudio.com/api/references/vscode-api#TextEdit)
could have been
delete(range: [Range](https://code.visualstudio.com/api/references/vscode-api#Range) | [Selection](https://code.visualstudio.com/api/references/vscode-api#Selection)): [TextEdit](https://code.visualstudio.com/api/references/vscode-api#TextEdit)
replace(range: [Range](https://code.visualstudio.com/api/references/vscode-api#Range) | [Selection](https://code.visualstudio.com/api/references/vscode-api#Selection), newText: string): [TextEdit](https://code.visualstudio.com/api/references/vscode-api#TextEdit)
respectively.

**Comment by alefragnani** [comment, marked as answer]
Hi @kedicesur ,

I'm not sure what are you trying to achieve that makes you notice the API could be improved. In fact, I'm not that familiar with the `TextEdit` API, and how it compares and/or combines with `TextEditorEdit` (which has `Selection` as alternative input). But, if are just making a feature request, I suggest you to use the the VS Code repository instead (https://github.com/microsoft/vscode/issues/new/choose). This Discussions repo is focused for extension development for now.

On the other hand, if you are having any issue with the APIs and would like do discuss how to accomplish what your extension needs, please, provide more details and we could try to help you with.

Hope this helps
